vendorpromo/config.py

```diff
@@ -1,6 +1,70 @@
 # App Settings
 from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.sites.models import Site
+from django.utils.translation import gettext_lazy as _
 
+from siteconfigs.config import SiteConfigBaseClass
+from siteconfigs.models import SiteConfigModel
+from vendorpromo.forms import PromoProcessorForm, PromoProcessorSiteSelectForm, SupportedPromoProcessor
+
+
+class PromoProcessorSiteConfig(SiteConfigBaseClass):
+    label = _("Promo Code Processor")
+    default = {"promo_processor": "base.PromoProcessorBase"}
+    form_class = PromoProcessorForm
+    key = ""
+    instance = None
+
+    def __init__(self, site=None):
+        if site is None:
+            site = Site.objects.get_current()
+        self.key = ".".join([__name__, __class__.__name__])
+        self.set_instance(site)
+        super().__init__(site, self.key)
+
+    # TODO: This should be implemented in the SiteConfigBaseClass  
+    def save(self, valid_form):
+        site_config, created = SiteConfigModel.objects.get_or_create(site=self.site, key=self.key)
+        site_config.value = {"promo_processor": valid_form.cleaned_data["promo_processor"]}
+        site_config.site = self.site
```

Comment:
not necessary to declare site here, since it's already specified by the get_or_create

Reply:
agree